### PR TITLE
state object flag

### PR DESCRIPTION
**Pull request by stevenyangek9**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/state_object.go [hunk 1]
```diff
@@ -84,12 +101,24 @@ func newStateObject(db *CommitStateDB, accProto authexported.Account) *stateObje
 		panic(fmt.Sprintf("invalid account type for state object: %T", accProto))
 	}
 
+	if useCache(db) {
```
Comment (KamiD) [inline x/evm/types/state_object.go:102]: if it running under mode of CheckTx, how to update the global cache, how to handle cache conflict

File: x/evm/types/statedb.go [hunk 2]
```diff
@@ -869,6 +869,12 @@ func (csdb *CommitStateDB) updateStateObject(so *stateObject) error {
 
 // deleteStateObject removes the given state object from the state store.
 func (csdb *CommitStateDB) deleteStateObject(so *stateObject) {
+	if !so.stateDB.ctx.IsCheckTx() {
```
Comment (KamiD) [inline x/evm/types/statedb.go]: the option of Pushback was running without check of IsCheckTx, but delete option added

File: x/evm/types/state_object.go [hunk 3]
```diff
@@ -465,3 +507,10 @@ type stateEntry struct {
 	address     ethcmn.Address
 	stateObject *stateObject
 }
+
+func useCache(db *CommitStateDB) bool {
+	if GlobalStateObjectCacheSize > 0 && !db.ctx.IsCheckTx() && db.ctx.BlockHeight() > GlobalCacheBeginHeight {
```
Comment (KamiD) [inline x/evm/types/state_object.go]: to avoid smb, the params of height must hardcoded

File: x/evm/types/state_object.go (hunk 1 above)
Comment (stevenyangek9) [reply x/evm/types/state_object.go:102]: useCache  means  GlobalStateObjectCacheSize > 0 && !db.ctx.IsCheckTx() , so cache can't work under  mode of CheckTx, under deliverTX mod every tx is serial processing, the cache can't  conflict.

File: x/evm/types/statedb.go (hunk 2 above)
Comment (stevenyangek9) [reply x/evm/types/statedb.go]: this is fix by using useCache function to judge

**Pull request closed without merging**